"Focus" the current unmodified upstream template in such a way that all
uncommented entries are copied to the end of the file, while they are
commented out by "###" in the original places.

Index: squid/squid.conf
===================================================================
--- squid.orig/squid.conf
+++ squid/squid.conf
@@ -639,32 +639,32 @@
 #
 # Recommended minimum configuration:
 #
-acl manager proto cache_object
-acl localhost src 127.0.0.1/32 ::1
-acl to_localhost dst 127.0.0.0/8 0.0.0.0/32 ::1
+###acl manager proto cache_object
+###acl localhost src 127.0.0.1/32 ::1
+###acl to_localhost dst 127.0.0.0/8 0.0.0.0/32 ::1
 
 # Example rule allowing access from your local networks.
 # Adapt to list your (internal) IP networks from where browsing
 # should be allowed
-acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
-acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
-acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
-acl localnet src fc00::/7       # RFC 4193 local private network range
-acl localnet src fe80::/10      # RFC 4291 link-local (directly plugged) machines
-
-acl SSL_ports port 443
-acl Safe_ports port 80		# http
-acl Safe_ports port 21		# ftp
-acl Safe_ports port 443		# https
-acl Safe_ports port 70		# gopher
-acl Safe_ports port 210		# wais
-acl Safe_ports port 1025-65535	# unregistered ports
-acl Safe_ports port 280		# http-mgmt
-acl Safe_ports port 488		# gss-http
-acl Safe_ports port 591		# filemaker
-acl Safe_ports port 777		# multiling http
-acl Safe_ports port 901		# SWAT
-acl CONNECT method CONNECT
+###acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
+###acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
+###acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+###acl localnet src fc00::/7       # RFC 4193 local private network range
+###acl localnet src fe80::/10      # RFC 4291 link-local (directly plugged) machines
+
+###acl SSL_ports port 443
+###acl Safe_ports port 80		# http
+###acl Safe_ports port 21		# ftp
+###acl Safe_ports port 443		# https
+###acl Safe_ports port 70		# gopher
+###acl Safe_ports port 210		# wais
+###acl Safe_ports port 1025-65535	# unregistered ports
+###acl Safe_ports port 280		# http-mgmt
+###acl Safe_ports port 488		# gss-http
+###acl Safe_ports port 591		# filemaker
+###acl Safe_ports port 777		# multiling http
+###acl Safe_ports port 901		# SWAT
+###acl CONNECT method CONNECT
 
 #  TAG: follow_x_forwarded_for
 #	Allowing or Denying the X-Forwarded-For header to be followed to
@@ -763,14 +763,14 @@ acl CONNECT method CONNECT
 # Recommended minimum Access Permission configuration:
 #
 # Only allow cachemgr access from localhost
-http_access allow manager localhost
-http_access deny manager
+###http_access allow manager localhost
+###http_access deny manager
 
 # Deny requests to certain unsafe ports
-http_access deny !Safe_ports
+###http_access deny !Safe_ports
 
 # Deny CONNECT to other than secure SSL ports
-http_access deny CONNECT !SSL_ports
+###http_access deny CONNECT !SSL_ports
 
 # We strongly recommend the following be uncommented to protect innocent
 # web applications running on the proxy server who think the only
@@ -784,14 +784,14 @@ http_access deny CONNECT !SSL_ports
 # Example rule allowing access from your local networks.
 # Adapt localnet in the ACL section to list your (internal) IP networks
 # from where browsing should be allowed
-http_access allow localnet
-http_access allow localhost
+###http_access allow localnet
+###http_access allow localhost
 
 # Allow the localhost to have access by default
-http_access allow localhost
+###http_access allow localhost
 
 # And finally deny all other access to this proxy
-http_access deny all
+###http_access deny all
 
 #  TAG: adapted_http_access
 #	Allowing or Denying access based on defined access lists
@@ -1076,7 +1076,7 @@ http_access deny all
 #
 
 # Squid normally listens to port 3128
-http_port 3128
+###http_port 3128
 
 #  TAG: https_port
 #	Usage:  [ip:]port cert=certificate.pem [key=key.pem] [options...]
@@ -1876,7 +1876,7 @@ http_port 3128
 #
 
 # We recommend you to use at least the following line.
-hierarchy_stoplist cgi-bin ?
+###hierarchy_stoplist cgi-bin ?
 
 # MEMORY CACHE OPTIONS
 # -----------------------------------------------------------------------------
@@ -2599,7 +2599,7 @@ hierarchy_stoplist cgi-bin ?
 #
 
 # Leave coredumps in the first cache dir
-coredump_dir /var/cache/squid
+###coredump_dir /var/cache/squid
 
 # OPTIONS FOR FTP GATEWAYING
 # -----------------------------------------------------------------------------
@@ -2927,10 +2927,10 @@ coredump_dir /var/cache/squid
 #
 
 # Add any of your own refresh_pattern entries above these.
-refresh_pattern ^ftp:		1440	20%	10080
-refresh_pattern ^gopher:	1440	0%	1440
-refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
-refresh_pattern .		0	20%	4320
+###refresh_pattern ^ftp:		1440	20%	10080
+###refresh_pattern ^gopher:	1440	0%	1440
+###refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
+###refresh_pattern .		0	20%	4320
 
 #  TAG: quick_abort_min	(KB)
 #  TAG: quick_abort_max	(KB)
@@ -5596,3 +5596,43 @@ refresh_pattern .		0	20%	4320
 #Default:
 # max_filedescriptors 0
 
+
+# Actual configuration
+# ====================
+
+acl manager proto cache_object
+acl localhost src 127.0.0.1/32 ::1
+acl to_localhost dst 127.0.0.0/8 0.0.0.0/32 ::1
+acl localnet src 10.0.0.0/8	# RFC1918 possible internal network
+acl localnet src 172.16.0.0/12	# RFC1918 possible internal network
+acl localnet src 192.168.0.0/16	# RFC1918 possible internal network
+acl localnet src fc00::/7       # RFC 4193 local private network range
+acl localnet src fe80::/10      # RFC 4291 link-local (directly plugged) machines
+acl SSL_ports port 443
+acl Safe_ports port 80		# http
+acl Safe_ports port 21		# ftp
+acl Safe_ports port 443		# https
+acl Safe_ports port 70		# gopher
+acl Safe_ports port 210		# wais
+acl Safe_ports port 1025-65535	# unregistered ports
+acl Safe_ports port 280		# http-mgmt
+acl Safe_ports port 488		# gss-http
+acl Safe_ports port 591		# filemaker
+acl Safe_ports port 777		# multiling http
+acl Safe_ports port 901		# SWAT
+acl CONNECT method CONNECT
+http_access allow manager localhost
+http_access deny manager
+http_access deny !Safe_ports
+http_access deny CONNECT !SSL_ports
+http_access allow localnet
+http_access allow localhost
+http_access allow localhost
+http_access deny all
+http_port 3128
+hierarchy_stoplist cgi-bin ?
+coredump_dir /var/cache/squid
+refresh_pattern ^ftp:		1440	20%	10080
+refresh_pattern ^gopher:	1440	0%	1440
+refresh_pattern -i (/cgi-bin/|\?) 0	0%	0
+refresh_pattern .		0	20%	4320
